app/Http/Controllers/SearchController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\User;
+use Illuminate\Support\Facades\Auth;
+
+class SearchController extends Controller
+{
+    /*
+     * Returns query results based on the search request.
+     * Exludes the user who is currently authenticated from
+     * search results.
+     */
+    public function search(Request $req) {
+        if (isset($req->search)) {
+            $query = trim($req->search);
+            $search_users = User::where('username', 'like', '%' . $query . '%')
+                ->where('username', '!=', Auth::user()->username)
```

Comment:
the search query should also check if the string matches the first name and last name